paddle/framework/init.cc

```diff
@@ -48,13 +49,17 @@ void InitDevices() {
   places.emplace_back(platform::CPUPlace());
 
 #ifdef PADDLE_WITH_CUDA
-  int count = platform::GetCUDADeviceCount();
-  for (int i = 0; i < count; ++i) {
-    places.emplace_back(platform::CUDAPlace(i));
+  try {
+    int count = platform::GetCUDADeviceCount();
+    for (int i = 0; i < count; ++i) {
+      places.emplace_back(platform::CUDAPlace(i));
+    }
```

Comment:
The strict way to write is to put line54~56 outside `try..catch`.

Reply:
done